packages/flutterfire_ui/lib/src/auth/widgets/phone_input.dart

```diff
@@ -135,6 +135,7 @@ class PhoneInputState extends State<PhoneInput> {
         (element) =>
             element.countryCode == countryCode ||
             element.phoneCode == phoneCode,
+        orElse: () => countries.first,
```

Comment:
I'm not entirely sure what exactly this PR intends to fix. This block of code is wrapped with try/catch and the case where no country code is found is handled in catch block

Reply:
Except the catch failed to catch the error, at least on my end. and that was the bug fired.

Reply:
@lesnitsky I am not sure why you closed this without deep investigation, for example, an assumption that a catch will catch all errors is a bad assumption, then why the bad state was firing? I am a bit confused in this case.

Reply:
@richard457 there is no such scenario where the exception wouldn't be caught in the code above

```dart
try {
      final newItem = countries.firstWhere(
        (element) =>
            element.countryCode == countryCode ||
            element.phoneCode == phoneCode,
      );
      // ...
    } catch (_) {
      // ...
    }
```

I think you had "All exceptions" checked in vscode debugger